src/SFA.DAS.PR.Jobs/Services/NotificationTokenService.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Microsoft.Extensions.Options;
+using SFA.DAS.PR.Data.Common;
+using SFA.DAS.PR.Data.Entities;
+using SFA.DAS.PR.Data.Repositories;
+using SFA.DAS.PR.Jobs.Configuration;
+
+namespace SFA.DAS.PR.Jobs.Services;
+
+public interface INotificationTokenService
+{
+    Task<Dictionary<string, string>> GetEmailTokens(Notification notification, CancellationToken cancellationToken);
+}
+
+public class NotificationTokenService(IProvidersRepository _providersRepository, IAccountLegalEntityRepository _accountLegalEntityRepository, IOptions<NotificationsConfiguration> _notificationConfigurationOptions) : INotificationTokenService
+{
+    private const string ApprovalsAdd = "add";
+    private const string ApprovalsCannotAdd = "cannot add";
+    private const string RecruitCreate = "create";
+    private const string RecruitCreateAndPublish = "create and publish";
+    private const string RecruitCannotCreate = "cannot create";
+
+    public async Task<Dictionary<string, string>> GetEmailTokens(Notification notification, CancellationToken cancellationToken)
+    {
+        Dictionary<string, string> emailTokens = new();
+
+        switch (Enum.Parse(typeof(NotificationType), notification.NotificationType))
+        {
+            case NotificationType.Provider:
+                {
+                    Provider? provider = await _providersRepository.GetProvider(notification.Ukprn!.Value, cancellationToken);
+
+                    AccountLegalEntity? accountLegalEntity = await _accountLegalEntityRepository.GetAccountLegalEntity(notification.AccountLegalEntityId!.Value, cancellationToken);
+
+                    emailTokens = new()
+                    {
+                        { EmailTokens.ProviderPortalUrlToken, _notificationConfigurationOptions.Value.ProviderPortalUrl},
```

Comment:
The Email template merge field is called provideraccountsweb

Reply:
This is a constant named based on the config setting, so we know how it maps. The value of the constant matches the email template.